### DISCUSSION TITLE
MultiApp transfer AuxKernel object at Gauss points?

### DISCUSSION THREAD

**Discussion by rcontinuum** [idaholab/moose#19634] [Q&A Modules: General] [answered]
Hello, 
I want to define an AuxKernel elemental variable directly at the Gauss points of the elements. And then transfer this variable to another app in the multiapp context. Say, both meshes of the master app and the sub app have the same mesh with the same quadrature rule, so that their Gauss points coincides. Then such a transfer should be feasible and would not involve any "interpolation". 

Is that currently possible? Is their a trick to do that? Any help is appreciated. 

If I understand correctly, then MOOSE AuxKernel offers volume averaged elemental variables (only?). 
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step10_auxkernels.html#step-10-develop-an-auxkernel-object
"When the variable type is elemental, the computeValue() method averages the values over the element quadrature points (QPs) weighted by their Gaussian quadratures in proportion to the total volume, area, or length of the element." 

Thank you

**Comment by GiudGiud** [comment]
Hello

I think the trick would be to use a shape function for the elemental variable that is exact on the Gauss points.
Any of the ones in the family parameter on this page does the trick?
https://mooseframework.inl.gov/syntax/Variables/index.html

Guillaume

**Comment by rcontinuum** [comment, marked as answer]
Hello, 
would you be so kind and elaborate a bit more? 
Assume I have a mesh of QUAD4 elements, bilinear Lagrange shape functions. 
Typically one uses a 2x2 Gauss integration. With Gauss points located at ±0.577... in the isoparametric space which I assume is 
<img src="https://render.githubusercontent.com/render/math?math=-1 \le \{\xi_1, \xi_2\} \le 1">

For this specific case which shape function would you use "for the elemental variable that is exact on the Gauss points".

Does there exists an example input file for this case or a similar case?

**Reply by GiudGiud** [reply]
If the master app and the subapp have the same mesh, you can use the copy transfer to move fields between apps
https://mooseframework.inl.gov/source/transfers/MultiAppCopyTransfer.html
if they locally have the same mesh, but in some places don't then you can use the projection transfer, which will still copy the field exactly if the mesh coincides locally. 
https://mooseframework.inl.gov/source/transfers/MultiAppProjectionTransfer.html

this will make sure the two fields have the same value at the Gauss points, since they have the same value everywhere. Please note that the order for the quadrature is chosen so that the product of the shape functions is integrated exactly.

But I think you want to be specifying the value at quadrature points directly as well. This is exactly what the auxkernel does. But when it returns the results, it doesnt return it on the quadrature point. It returns it on the nodes or on the elements. So there is an integration going on, with a system solve for higher orders, to go from quadrature points to nodal or elemental values.

**Comment by rcontinuum** [comment]
Thank you for the clarification!
Ralf
